web_src/js/features/repo-issue.js

```diff
@@ -226,22 +227,32 @@ export function initRepoIssueCodeCommentCancel() {
 export function initRepoPullRequestUpdate() {
   // Pull Request update button
   const $pullUpdateButton = $('.update-button > button');
-  $pullUpdateButton.on('click', function (e) {
+  $pullUpdateButton.on('click', async function (e) {
     e.preventDefault();
     const $this = $(this);
     const redirect = $this.data('redirect');
     $this.addClass('loading');
-    $.post($this.data('do'), {
-      _csrf: csrfToken
-    }).done((data) => {
-      if (data.redirect) {
-        window.location.href = data.redirect;
-      } else if (redirect) {
-        window.location.href = redirect;
-      } else {
-        window.location.reload();
-      }
-    });
+    let response;
+    try {
+      response = await POST($this.data('do'));
+    } catch (error) {
+      console.error('Error:', error);
+    } finally {
+      $this.removeClass('loading');
+    }
+    let data;
+    try {
+      data = await response?.json(); // the response is probably not a JSON
```

Comment:
Move line into above `try` imho.

Reply:
The response isn't a JSON though

Reply:
Can we find out what responses are possible?

Reply:
It returns the full page (HTML)

Reply:
Always? Then why do we try to parse JSON? :D

Reply:
I thought about it but the URL is dynamic and we don't replace content, we reload. It's certainly possible but I'd have to add some `form` and `inputs` and have the `HX-Refresh` header in the response so I decided it's not worth it